packages/frontend/src/redux/slices/tokens/index.js

```diff
@@ -76,6 +76,26 @@ const fetchTokens = createAsyncThunk(
     }
 );
 
+const fetchToken = createAsyncThunk(
+    `${SLICE_NAME}/fetchTokens`,
+    async ({ contractName, accountId }, thunkAPI) => {
+        const { dispatch, getState } = thunkAPI;
+        const { actions: { setContractMetadata } } = tokensSlice;
+        try {
+            const contractMetadata = await getCachedContractMetadataOrFetch(contractName, getState());
+            if (!selectOneContractMetadata(getState(), { contractName })) {
+                dispatch(setContractMetadata({ contractName, metadata: contractMetadata }));
+            }
+            if(accountId) {
+                await dispatch(fetchOwnedTokensForContract({ accountId, contractName }));
+            }
+        } catch (e) {
+            // Continue loading other likely contracts on failures
+            console.warn(`Failed to load FT for ${contractName}`, e);
+        }
+    }
+);
```

Comment:
I think that this thunk should have a `condition` argument provided to it with the same code as the `fetchOwnedTokensForContract` thunk does, so that we don't re-fetch contract metadata for a contract that we are already fetching it for.

Reply:
As discussed, an issue will be created for this since it requires changes to the state tree